paper-toggle-button.html

```diff
@@ -15,165 +20,143 @@
-<template>
+<dom-module id="paper-toggle-button">
 
-  <link rel="stylesheet" href="paper-toggle-button.css">
+  <style is="x-style">
+    * {
+      --paper-toggle-button-unchecked-bar-color: #000000;
+      --paper-toggle-button-unchecked-button-color: #f1f1f1;
+      --paper-toggle-button-unchecked-ink-color: #bbb;
+      --paper-toggle-button-checked-bar-color: #0f9d58;
+      --paper-toggle-button-checked-button-color: #0f9d58;
+      --paper-toggle-button-checked-ink-color: #0f9d58;
+    }
+  </style>
 
-  <core-a11y-keys target="{{}}" keys="space" on-keys-pressed="{{tap}}"></core-a11y-keys>
+  <link rel="import" type="css" href="paper-toggle-button.css">
 
-  <div id="toggleContainer" checked?="{{checked}}" disabled?="{{disabled}}">
-  
-    <div id="toggleBar" class="toggle toggle-bar"></div>
+  <template>
 
-    <div id="toggleButton" class="toggle toggle-button">
-      <paper-ripple id="ink" class="toggle-ink circle"></paper-ripple>
+    <div id="toggleContainer">
+      <div id="toggleBar" class="toggle-bar"></div>
+      <div id="toggleButton" class="toggle-button">
+        <paper-ripple id="ink" class="toggle-ink circle" recenters></paper-ripple>
+      </div>
     </div>
 
-  </div>
-
-</template>
-<script>
-
-  Polymer('paper-toggle-button', {
-    
-    /**
-     * Fired when the checked state changes due to user interaction.
-     *
-     * @event change
-     */
-     
-    /**
-     * Fired when the checked state changes.
-     *
-     * @event core-change
-     */
-
-    /**
-     * Gets or sets the state, `true` is checked and `false` is unchecked.
-     *
-     * @attribute checked
-     * @type boolean
-     * @default false
-     */
-    checked: false,
-
-    /**
-     * If true, the toggle button is disabled.  A disabled toggle button cannot
-     * be tapped or dragged to change the checked state.
-     *
-     * @attribute disabled
-     * @type boolean
-     * @default false
-     */
-    disabled: false,
-
-    eventDelegates: {
-      down: 'downAction',
-      up: 'upAction',
-      tap: 'tap',
-      trackstart: 'trackStart',
-      trackx: 'trackx',
-      trackend: 'trackEnd'
-    },
-
-    downAction: function(e) {
-      var rect = this.$.ink.getBoundingClientRect();
-      this.$.ink.downAction({
-        x: rect.left + rect.width / 2,
-        y: rect.top + rect.height / 2
-      });
-    },
-
-    upAction: function(e) {
-      this.$.ink.upAction();
-    },
-
-    tap: function() {
-      if (this.disabled) {
-        return;
-      }
-      this.checked = !this.checked;
-      this.fire('change');
-    },
-
-    trackStart: function(e) {
-      if (this.disabled) {
-        return;
+  </template>
+
+  <script>
+    Polymer({
+      is: 'paper-toggle-button',
+
+      behaviors: [
+        Polymer.PaperButtonBehavior
+      ],
+
+      // The custom properties shim is currently an opt-in feature.
+      enableCustomStyleProperties: true,
+
+      hostAttributes: {
```

Comment:
Guilty of that myself. They should probably be set in `ready`. WDYT about a behavior that helps manage this?